cargo-dist/src/announce.rs

```diff
@@ -173,20 +177,25 @@ fn check_dist_package(
     pkg: &axoproject::PackageInfo,
     announcing: &PartialAnnouncementTag,
 ) -> Option<String> {
-    let package_cdylibs = graph
+    let package_libraries = graph
         .package_metadata(pkg_id)
-        .package_cdylibs
-        .unwrap_or(false);
+        .package_libraries
+        .clone()
+        .unwrap_or_default();
 
-    let package_empty = if package_cdylibs {
-        pkg.binaries.is_empty() && pkg.cdylibs.is_empty()
-    } else {
-        pkg.binaries.is_empty()
-    };
+    let mut conditions = vec![pkg.binaries.is_empty()];
+    if package_libraries.contains(&LibraryStyle::CDynamic) {
+        conditions.push(pkg.cdylibs.is_empty());
+    }
+    if package_libraries.contains(&LibraryStyle::CStatic) {
+        conditions.push(pkg.cstaticlibs.is_empty());
```

Comment:
I was very enamoured with how clever this was but just realized you can replace every `push` with just `&=`, which is probably clearer/simpler?